### ISSUE TITLE
Add an onboarding tour 

### ISSUE THREAD

**Issue by lTimeless** [SmartHub-Io/SmartHub#28]
The first time a user logs in he is getting an onboarding tour of the system. 
It shows him where he can find stuff and how to edit it in a modal like manner.

Notes:
- Different roles could get some longer/shorter tours.
- There are some Js packages out there that could be useful

**Comment by davidsvk** [comment]
Hi, could I work on this issue? @lTimeless

**Comment by lTimeless** [comment]
@davidsvk Sure, you have an idea what npm package you wanna use ??
And if you have any questions, feel free to ask anything on [discord](https://discord.com/channels/760946711794679819/760946713024004175) ;)

**Comment by davidsvk** [comment]
Yes, maybe vue-tour package, it looks pretty good.
Thank you :). @lTimeless

**Comment by lTimeless** [comment]
@davidsvk That package look really awesome :) 
But i couldn't find support for Vue3 maybe that could get a problem, just a heads up.

**Comment by davidsvk** [comment]
@lTimeless Yes, you were right. Unfortunatelly, the current version of vue-tour package does not work with Vue3. But I did some research to find out differences between the older Vue2 and newer Vue3. 

First significant difference is in using a new global API. So it changes the way, how the app is created and how new packages/libraries are instaled into it. So firstly, I edited a few lines of code in vue-tour package (main.js) to adjust package to this new API.

Also, I had to make a few more changes, because of new Vue3 composition API, which you are using. Basicly, it is not possible to use "this" reference with it. So in order to access (read/write) global properties I adjusted code in package by using inject and provide.

And it is working now :). 
I will test if everything is running how it should and then I will start making the onboarding tour.

**Comment by lTimeless** [comment]
@davidsvk Good to hear, I'm curious about the result. 😀
If you are making changes to their code, maybe you should make a PR too that project aswell 🤔

And sorry for the big changes on the develop branch lately.

**Comment by davidsvk** [comment]
@lTimeless No problem :), I had some work to complete at school, so I didn't have time to look at the issue last week. So today I finally got to it and wanted to finish it. I updated the branch to the latest, but I have trouble running the backend now. It does not turn on and ends with an error. Can you help me?
[log.txt](https://github.com/SmartHub-Io/SmartHub/files/5460695/log.txt)

**Comment by lTimeless** [comment]
@davidsvk I've found the mistake... The last PR I accepted had some issues... The project builds so CI wasn't throwing an error but for ef-core every Entity need to have a parameterless constructor and they got removed with the last PR.

With my PR #45 it will be fixed. It will get merged asap.

**Comment by lTimeless** [comment]
@davidsvk The fix is now on the develop branch. Please try it out an dif oyu find any Issues let me know

**Comment by davidsvk** [comment]
Thank you for your help.

I tried to run it, but unfortunately, another error appeared (with creating "Homes" table). The backend is not working properly then (every request ends with a 500 status code).

Logfile:
[log.txt](https://github.com/SmartHub-Io/SmartHub/files/5466151/log.txt)

**Comment by lTimeless** [comment]
@davidsvk drop your database and than update it again. That should hopefully do the trick.

**Comment by davidsvk** [comment]
So I finished the task. @lTimeless 
As I said, I was first working on the package code itself. So now it is compatible with the latest version of Vue3 and it is able to use the new composition API. Then I also decided to modify the package design a little bit to match the visual part of your app. 

Finally, I created an onboarding tour by using that modified package.
Currently, the onboarding tour will start every time when you log into the app. If you want I can use local storage for storing information that the tour was already finished.

So you can take a look now :).
If you have any questions or suggestions, please let me know.

**Comment by lTimeless** [comment]
@davidsvk 
I will lookat it later ;)

Maybe add a Property to the userEntity in the DB instead of saving it into localstorage. This way it will not be resetet after the browser cache gets cleared.
